test.sh

```diff
@@ -205,22 +209,28 @@ if [ "$1" == "spark" ]; then
         exit 0
     fi
 
+    shift
     spark_setup
     echo "Running tests"
     docker-compose -f $SPARK_COMPOSE_FILE_LOC \
                    -p $SPARK_COMPOSE_PROJECT_NAME \
-                run --rm request_consumer
+                run \
+                -e PYTHONDONTWRITEBYTECODE=1 \
+                --rm request_consumer \
+                dockerize -wait tcp://namenode:9000 -timeout 60s \
+                python -m pytest -c pytest.spark.ini "$@"
     RET=$?
     spark_dcdown
     exit $RET
 fi
 
 if [ "$1" == "int" ]; then
+    if [ "$2" == "-b" ]; then
+        int_build
+        exit $?
+    fi
     echo "Taking down old containers"
     int_dcdown
-    echo "Building current setup"
```

Comment:
one nice thing about the original code is that you can just run `./test.sh int` and it'll take care of everything - running a build if necessary and then doing tests. By moving it out into `-b` we have to remember to periodically run build if we change dependencies, etc. If you don't want to run a build each time during CI tests, I think my preference here would be to have a specific workflow for CI which is different to the developer interface

Reply:
Yeah, I was thinking this too. Currently, the sharding and artifact dance does not seem to improve tests run as much as I would have thought. I wondering if its worth to pursue this PR currently. This will probably be useful when we merge unit/integration tests but as things stand currently the gain in tests speed probably do not justify adding more stuff around this.